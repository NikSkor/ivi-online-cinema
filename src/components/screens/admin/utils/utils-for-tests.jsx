import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import adminReducer from '../../../../store/slices/adminSlice';


export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { admin: adminReducer},
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}