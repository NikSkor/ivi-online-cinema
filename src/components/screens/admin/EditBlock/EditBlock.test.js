import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import EditBlock from '@/components/screens/admin/EditBlock/EditBlock';
import { renderWithProviders } from '../utils/utils-for-tests';

jest.mock('next/router', () => require('next-router-mock'));

describe('test EditBlock component', () => {
  afterEach(cleanup);

  test('component renders props correctly, truthy values', () => {
    const tree = renderWithProviders(
      <EditBlock
        titleName={'Название'}
        name={'Жанр'}
        enName={'Genre'}
        getName={() => {}}
        getEnName={() => {}}
      />
    );
    expect(tree).toMatchSnapshot();
  });
  test('component renders props correctly, falsy values', () => {
    const tree = renderWithProviders(<EditBlock />);
    expect(tree).toMatchSnapshot();
  });
});
