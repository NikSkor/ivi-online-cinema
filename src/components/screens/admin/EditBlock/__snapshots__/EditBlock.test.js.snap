// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test EditBlock component component renders props correctly, falsy values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="main"
      >
        <h2
          class="title"
        />
        <div
          class="form"
        >
          <label
            class="label"
          >
            Title:
            <input
              class="inputs"
              placeholder="Enter the title"
              test-id="input"
              type="text"
              value=""
            />
          </label>
          <label
            class="label"
          >
            Title in English::
            <input
              class="inputs"
              placeholder="Enter English title"
              test-id="input"
              type="text"
              value=""
            />
          </label>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="main"
    >
      <h2
        class="title"
      />
      <div
        class="form"
      >
        <label
          class="label"
        >
          Title:
          <input
            class="inputs"
            placeholder="Enter the title"
            test-id="input"
            type="text"
            value=""
          />
        </label>
        <label
          class="label"
        >
          Title in English::
          <input
            class="inputs"
            placeholder="Enter English title"
            test-id="input"
            type="text"
            value=""
          />
        </label>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`test EditBlock component component renders props correctly, truthy values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="main"
      >
        <h2
          class="title"
        >
          Название
        </h2>
        <div
          class="form"
        >
          <label
            class="label"
          >
            Title:
            <input
              class="inputs"
              placeholder="Enter the title"
              test-id="input"
              type="text"
              value="Жанр"
            />
          </label>
          <label
            class="label"
          >
            Title in English::
            <input
              class="inputs"
              placeholder="Enter English title"
              test-id="input"
              type="text"
              value="Genre"
            />
          </label>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="main"
    >
      <h2
        class="title"
      >
        Название
      </h2>
      <div
        class="form"
      >
        <label
          class="label"
        >
          Title:
          <input
            class="inputs"
            placeholder="Enter the title"
            test-id="input"
            type="text"
            value="Жанр"
          />
        </label>
        <label
          class="label"
        >
          Title in English::
          <input
            class="inputs"
            placeholder="Enter English title"
            test-id="input"
            type="text"
            value="Genre"
          />
        </label>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
